backends/ebpf/CMakeLists.txt

```diff
@@ -105,19 +105,24 @@ endif()
 
 # Check if we have the right llvm version
 set (MIN_LLVM 3.7.1)
-
-# If you get an error here see
-# https://stackoverflow.com/questions/38171543/error-when-using-cmake-with-llvm
-find_package(LLVM QUIET)
-if (LLVM_PACKAGE_VERSION)
+# Grab the LLVM version, do not use find_package because it is unstable
+# https://github.com/p4lang/p4c/issues/1376
+set(LLVM_CMD "llvm-config --version")
+message(STATUS "Check LLVM version with '${LLVM_CMD}'")
+exec_program(${LLVM_CMD}
+	 RETURN_VALUE LLVM_RET
+	 OUTPUT_VARIABLE LLVM_PACKAGE_VERSION)
+if (NOT LLVM_RET)
   message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
   if (${LLVM_PACKAGE_VERSION} VERSION_LESS ${MIN_LLVM})
-    MESSAGE(WARNING "LLVM version ${LLVM_PACKAGE_VERSION} too small, expected ${MIN_LLVM}.
+    message(WARNING "LLVM version ${LLVM_PACKAGE_VERSION} too small, expected ${MIN_LLVM}.
     Ignoring ebpf tests...")
     set (SUPPORTS_KERNEL False)
   endif()
 else()
-  message(STATUS "LLVM missing, disabling kernel tests...")
+  message(WARNING
+	 "Problem with LLVM, disabling kernel tests...\n"
```

Comment:
This message could be more user-friendly.
Perhaps "Could not detect a version of LLVM which is needed to compile the eBPF kernel tests".

Did you test this on both systems that have working and broken llvms?

Reply:
Yes, I tested it on an Ubuntu Xenial image with the broken llvm package and on my local version. Both worked fine. The Travis test also passed. This method does not seem to work on macOS, but we do not support that anyway as far as I remember.

Reply:
On Mac we don't support eBPF, but the build should still work, of course.
Travis will test on Mac, so I think that if that passes (and it did previously) then we can merge this.

Reply:
Okay, it should pass, it will just complain that it cannot find llvm, which also might be just a setup issue in the macOS Travis image.